caktus_primer/static/js/modal.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+
+let clickEvents = ['click', 'mousedown', 'touchstart', 'dblclick']
+
+
+class Modal {
+    constructor(id) {
+        this.name = 'Modal'
+        this.options = {}
+        this.container = document.getElementById(id) || null
+        if (this.container) {
+            this.openTarget = this.container.querySelector('[aria-label="open"]')
+            this.dialogTarget = this.container.getElementsByTagName('dialog')[0]
+            this.closeTarget = this.container.querySelector('[aria-label="close"]')
+        }
+        else {
+            return
+        }
+
+        clickEvents.forEach((n) => (this.openTarget.addEventListener(n, (e) => { this.handleClick(e)})))
+        
+        clickEvents.forEach((n) => (this.closeTarget.addEventListener(n, (e) => { this.closeModal(e)})))
+        
+        document.addEventListener('keypress', (e) => { 
+            // Escape key
+            if (e.keyCode == 27) {
+                this.closeModal(e) 
+            }
+        })
+    }
+
+    handleClick() {
+        this.dialogTarget.showModal()
+        document.body.style.overflow = 'hidden';
+    }
+
+    closeModal() {
+        this.dialogTarget.close()
+        document.body.style.overflow = '';
+    }
+
+}
+
+class LargeModal extends Modal {
+    constructor(id) {
+        super(id)
+        this.options['size'] = '80%'
+        this.name = 'LargeModal'
+        this.dialogTarget.style.width = this.options['size']
+    }
+}
+
+class MediumModal extends Modal {
+    constructor(id) {
+        super(id)
+        this.options['size'] = '50%'
+        this.name = 'MediumModal'
+        this.dialogTarget.style.width = this.options['size']
+    }
+}
+
+class SmallModal extends Modal {
+    constructor(id) {
+        super(id)
+        this.options['size'] = '30%'
+        this.name = 'MediumModal'
+        this.dialogTarget.style.width = this.options['size']
+    }
+}
```

Comment:
Shouldn't these issues of sizes, being styling, exist as part of the stylus and not be set in JS?

Reply:
The intended use case of this Modal module is to let users instantiate 3 different modals based on which size they'd like. The code is set up to accept an ID attribute and adds the needed functionality and styling.

Reply:
I really don't think separate JS classes for different sizes are necessary. If the only difference is the one option (the `name` isn't used I notice) you could let that option be passed when creating the `Modal` instance, so just pass the size as a parameter instead of using different classes. That'll mean less code to maintain, and you can use different sizes when its called for.

If you still want to have the three named sizes (rather than just pass the `%` manually, maybe you could let that option accept `"small"`, `"medium"`, and `"large"` and map them to the sizes, but it should definitely be the single `Modal` class you already have.